packages/core/src/lib/store/index.ts

```diff
@@ -220,6 +229,23 @@ class Store extends BaseProtocol implements IStore {
       endTime = options.timeFilter.endTime;
     }
 
+    // convert array to set to remove duplicates
+    const uniquePubSubTopicsInQuery = Array.from(
+      new Set(decoders.map((decoder) => decoder.pubSubTopic))
+    );
+
+    // If multiple pubsub topics are provided, throw an error
+    if (uniquePubSubTopicsInQuery.length > 1) {
+      throw new Error(
+        "API does not support querying multiple pubsub topics at once"
+      );
+    }
+
+    // we can be certain that there is only one pubsub topic in the query
```

Comment:
I think we cannot because it might be `uniquePubSubTopicsInQuery` is empty.

 🙏  let's handle that case as we already saw errors caused by such a mistake.

Reply:
will make a PR with a change

Reply:
https://github.com/waku-org/js-waku/pull/1612

Reply:
I agree that it works now - no guarantee it will continue to work later.
There is no strict bound between those two parts of the code.